src/core/events.ts

```diff
@@ -4,6 +4,25 @@
 export enum MessageEvents {
   /** Fires when a new chat message is received. */
   Created = 'message.created',
+
+  /** Fires when a chat message is updated. */
+  Updated = 'message.updated',
+
+  /** Fires when a chat message is deleted. */
+  Deleted = 'message.deleted',
```

Comment:
If we support hard deleting in the future, do we need to differentiate here?

Reply:
I'm not too sure, and think that will depend on how it gets implemented in Ably. If it looks the same as a regular deletion message, just with the string cleansed etc.. my gut says no, that we should handle that within the message as part of the operation field, maybe adding it to the metadata object that gets returned, or as a totally separate ably specific field.

Reply:
Considering we aren't quite sure what a hard delete will look like, I'd say we stick with what we have for now.